news/2017-12-14-weld-meets-junit5

```diff
@@ -0,0 +1,101 @@
+---
+layout: news
+title: Weld meets JUnit 5
+author: Matej Novotny
+priority: 1
+change_frequency: weekly
+excerpt: Weld meets JUnit 5
+desc: What if you could easily write JUnit 5 tests for Weld beans?
+tags: [junit, testing, weld-se]
+---
+:linkattrs:
+:toc: macro
+:toc-title: Contents
+:icons: font
+
+toc::[]
+
+In this article I want to bring link:https://github.com/weld/weld-junit[weld-junit, window="_blank"] into light.
+You might have heard about or, better still, used our extension for JUnit 4 which has been around for some time now.
+But after getting ourselves acquainted with JUnit 5, we decided we should create another one for JUnit 5.
+In the sections below I will try to introduce the basic usage and ideas behind it, for more in depth information, you can glance the link:https://github.com/weld/weld-junit/blob/master/junit5/README.md[readme file] residing directly in the repository.
+
+== GAV Coordinates
+
+Here are group and artifact IDs you need to grab the artifact for Maven projects:
+
+----
+<groupId>org.jboss.weld</groupId>
+<artifactId>weld-junit-parent</artifactId>
+<version>${weld.junit.version}</version>
+----
+
+== First Steps
+
+You can use the extension as you would any other in JUnit 5, via `@ExtendWith` from JUnit library along with the name of our extension - `org.jboss.weld.junit5.WeldJunit5Extension`.
+Alternatively, there is a shorter annotation `@org.jboss.weld.junit5.EnableWeld`.
+Both of these can be used on class and method level.
+
+Hence your test could look like this:
+
+[ source, java ]
+----
+@EnableWeld
+public class InjectionWithWeldTest {
+
+  @Inject
+  BarBean beanAsField;
+
+  @Test
+  public void testThatItWorks(FooBean beanAsParam) {
+    assertNotNull(beanAsParam);
+    beanAsParam.ping();
+    assertNotNull(beanAsField);
+    beanAsField.ping();
+  }
+
+}
+----
+
+Behind the scenes, Weld will bootstrap SE container for each test container invocation and tear it down afterwards (works for both, per-method and per-class lifecycle).
+As you might have guessed, Weld will also automatically inject into any fields within the test class which you annotated with `@Inject`.
+Furthermore, it will attempt to resolve parameters in your test methods in very much the same way.
+You can then use the beans to your liking.
```

Comment:
I would add a note about the fact that only classes from within the test package are scanned by default.